kernel/tests/read.rs

```diff
@@ -404,15 +404,20 @@ fn read_with_scan_data(
         for read_result in read_results {
             let read_result = read_result.unwrap();
             let len = read_result.len();
-
-            // ask the kernel to transform the physical data into the correct logical form
-            let logical = transform_to_logical(
-                engine,
-                read_result,
-                &global_state,
-                &scan_file.partition_values,
-            )
-            .unwrap();
+            // to transform the physical data into the correct logical form
+            let logical = if let Some(ref transform) = scan_file.transform {
```

Comment:
I think this is (at least) three places that do ~exactly the same thing. Is there a way to factor out a helper method that everyone can use?